src/test/java/org/prebid/server/handler/openrtb2/AmpHandlerTest.java

```diff
@@ -443,6 +444,62 @@ public void shouldRespondWithCustomTargetingIncluded() {
                 + "\"hb_cache_id_bidder1\":\"value2\"}}"));
     }
 
+    @Test
+    public void shouldRespondWithAdditionalTargetingIncludedWhenSeatBidExists() {
+        // given
+        given(ampRequestFactory.fromRequest(any(), anyLong()))
+                .willReturn(Future.succeededFuture(givenAuctionContext(identity())));
+
+        final Map<String, JsonNode> targeting = new HashMap<>();
+        targeting.put("key", TextNode.valueOf("value"));
+        targeting.put("test-key", TextNode.valueOf("test-value"));
+
+        givenHoldAuction(BidResponse.builder()
+                .ext(ExtBidResponse.builder().additionalTargeting(targeting).build())
+                .seatbid(singletonList(SeatBid.builder()
+                        .seat("bidder1")
+                        .bid(singletonList(Bid.builder()
+                                .ext(mapper.valueToTree(
+                                        ExtPrebid.of(
+                                                ExtBidPrebid.builder().build(),
+                                                mapper.createObjectNode())))
+                                .build()))
+                        .build()))
+                .build());
```

Comment:
Please, consider extracting some parts of this huge call chains to local variables to reduce nesting.